src/listing.rs

```diff
@@ -219,10 +219,10 @@ pub fn directory_listing(
 
     // If the `qrcode` parameter is included in the url, then should respond to the QR code
     if let Some(url) = query_params.qrcode {
-        let res = match QrCode::encode_text(&url, QrCodeEcc::Medium) {
```

Comment:
Looks like someone was in a hurry and overlooked this. `url` is not actually the URL, so we're basically encoding an arbitrary query value.